modules/iam-group-with-policies/policies.tf

```diff
@@ -43,7 +43,7 @@ data "aws_iam_policy_document" "iam_self_management" {
     resources = [
       "arn:${local.partition}:iam::${local.aws_account_id}:user/*/$${aws:username}",
       "arn:${local.partition}:iam::${local.aws_account_id}:user/$${aws:username}",
-      "arn:${local.partition}:iam::${local.aws_account_id}:mfa/$${aws:username}",
```

Comment:
I think we should make this a variable where the default is the current behavior, but users can override with a wildcard for multiple MFA devices

Reply:
I will take another look to confirm (create a new IAM user and try to add a virtual MFA device with current `IAMSelfManagement` policy).
But it seems to me that since this new feature was released, the very first MFA device resource ARN take the name defined by user during his creation.

Reply:
To have a more scoped policy with no impacts, I made few changes : 
- Rollback previous changes (`AllowSelfManagement` & `AllowDeactivateMFADevice` statements)
- Add 2 more explicit statements :
  - `AllowManageOwnVirtualMFADevice` => to allow action `iam:CreateVirtualMFADevice` with the correct resource name
  - `AllowDeleteVirtualMFADevice` => to allow the action with the correct resource name

Reply:
Hello, 

I made few changes here, according to [this AWS documentation ](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html). 

The AWS policy allows IAM users **that are authenticated using multi-factor authentication (MFA)** to manage their own credentials on the _My security credentials page_. It also **requires** the user to set up and authenticate using MFA before performing **any** other operations in AWS ! 

In real terms, that enforce MFA utilization. Because the IAM user **cannot do anything** before his MFA is configured, and he is authenticated with. 

This is a great security improvement but this is a big change too. Is it what we want ?

Reply:
@bryantbiggs Are you OK with that ? 

If you're not, I can try to add a new variable to allow the user to choose between the [standard policy](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage-no-mfa.html) (which do not force MFA utilization) or the [more secure](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws_my-sec-creds-self-manage.html) (which force MFA).

Reply:
yes, I think this looks great and a good improvement; thank you